packages/blade/src/components/Tag/__tests__/Tag.native.test.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { fireEvent } from '@testing-library/react-native';
+import { Tag } from '../Tag';
+import renderWithTheme from '~utils/testing/renderWithTheme.native';
+
+describe('<Tag />', () => {
+  it('should render tag', () => {
+    const { toJSON } = renderWithTheme(
+      <Tag
+        onDismiss={() => {
+          console.log('tag dismissed');
+        }}
+      >
+        in:User
+      </Tag>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should call onDismiss', () => {
+    const dismissHandler = jest.fn();
+    const { queryAllByText, getAllByLabelText } = renderWithTheme(
+      <Tag onDismiss={dismissHandler}>in:User</Tag>,
+    );
+    expect(queryAllByText('in:User')[1]).toBeTruthy();
+    fireEvent.press(getAllByLabelText('Close in:User tag')[1]);
+    expect(dismissHandler).toBeCalledWith();
```

Comment:
assert that the `in:User` is falsy as well since it shall be dismissed?

Reply:
It won't be falsy actually. We expect consumers to handle the closing state no so when dismiss is called, consumer removes the tag. From our end for unit test, just calling onDismiss handles the cases.

Reply:
but we shall have a story that captures hiding of tag as well right?

Reply:
I tried this but its not really testing any additional behaviour apart from React's useState. Feels like too much mocking for this usecase in particular. In something like [Dropdown we test the usage behaviour](https://github.com/razorpay/blade/blob/master/packages/blade/src/components/Dropdown/__tests__/Dropdown.web.test.tsx#L398-L423) because it deals with our internal props as well such as `value` `onChange` etc. Which is not the case with Tag.

<img width="802" alt="image" src="https://github.com/razorpay/blade/assets/30949385/7de9b0a3-bba1-4a3e-9212-eb4472342090">